### DISCUSSION TITLE
Is it ok to use EVP_MD_* within EVP_cipher?

### DISCUSSION THREAD

**Discussion by rajeev-0** [openssl/openssl#22962] [Q&A] [answered]
I am implementing Integrity only cipher (RFC9150) https://github.com/openssl/openssl/pull/22903, which use HMAC for authentication and integrity. 
Can I use `HMAC_CTX` within cipher implementation (which implicitly use EVP_MD) or should I implement HMAC code within cipher file?

**Comment by t8m** [comment, marked as answer]
You should not add your own implementation of HMAC. So yeah, you can use HMAC_CTX internally from within the provider implementation of this stitched cipher.
